source/localizable/routing/defining-your-routes.md

```diff
@@ -29,6 +29,16 @@ Router.map(function() {
 });
 ```
 
+If your route name has multiple works then your route should follow convention and be hyphenated. Routes with underscores in them will not be rendered.
```

Comment:
> It's hard to explain in a way that doesn't talk about the resolver/container

I think the easiest explanation is just about files on disk. ember-cli expects all files on disk to be dasherized.  The first argument to `this.route` represents the path on disk so it should be dasherized.  If however, you need something else in your URL's you can use `path` option to customize...